src/Message/TransactionSearchResponse.php

```diff
@@ -12,9 +12,14 @@ public function __construct(RequestInterface $request, $data)
         parent::__construct($request, $data);
 
         $transactions = array();
+
         if ($this->isSuccessful()) {
-            foreach ($this->getData()['transactions']['transaction'] as $transaction) {
-                $transactions[] = $this->xml2array($transaction);
+            if ($this->getData()['resultsInThisPage'] == 1) {
```

Comment:
I didn't understand why did you change this part of the code. Can you explain it a little bit, please?

Reply:
If `transactions` returned is only 1, it is a single element. Therefore a `foreach` fails.